src/main/scala/rise/elevate/rules/traversal.scala

```diff
@@ -69,12 +69,24 @@ object traversal {
     implicit object RiseTraversable extends implementation.DefaultTraversal {
       override protected def oneHandlingState: Boolean => Strategy[Rise] => Strategy[Rise] =
         carryOverState => s => {
-          // (option 1) traverse to argument first
-          case a @ App(f, e) => s(e) match {
-            case Success(x: Rise) => Success(App(f, x)(a.t))
-            case Failure(state)   => if (carryOverState)
-              state(f).mapSuccess(App(_, e)(a.t)) else
-              s(f).mapSuccess(App(_, e)(a.t))
+          // To achieve a traversal that most closely corresponds to the execution order we ...
```

Comment:
can you explain why we would want the traversal order to be closer to execution order?

Reply:
Otherwise, traversals such as `outermost` and `innermost` have not the excepted effect.

Without the proposed changes `outermost(isMap)` will trigger the most nested `map` in the following expression instead of the one in the first line:
```
A |> map(fun(arow =>
  B |> transpose |> map(fun(bcol =>
    zip(arow)(bcol) |> map(multP) |> reduce(add)(0)
  )
)
```

Reply:
so if I traverse `f(g(v)) ~ v |> g |> f`, I will first traverse `f`, then `v`, then `g`? it is hard for me to see how this order fits the intuition of `outermost` or `innermost`, and more importantly why this should affect other traversals such as a simple `topDown` or `bottomUp`. What was the issue with the previous traversal order where we always visited the function first?